cloudsync/tasks.py

```diff
@@ -23,58 +27,100 @@
 THUMBNAIL_PATTERN = "thumbnails/{}_thumbnail_{{count}}"
 
 
-@shared_task(bind=True)
-def stream_to_s3(self, url, s3_key):
+class VideoTask(Task):
+    """
+    Custom Celery Task class for video uploads and transcodes
+    """
+
+    def get_task_id(self):
+        """
+        Get the task id (depending on whether the task is chained or not)
+
+        Args:
+            request(Task.request): The task request
+
+        Returns:
+            The request id
+        """
+        if self.request.chain:
+            try:
+                return self.request.chain[0]['options']['task_id']
+            except (IndexError, KeyError,) as exc:
+                # Log the error and continue, using self.request.id instead
+                # The worst that will happen is that progress bar won't work.
+                log.error("Could not find task_id in chain: %s", str(exc))
```

Comment:
this will make the function return `self.request.id` that I do not think it is what should do